docs/authdecode.md

```diff
@@ -0,0 +1,216 @@
+```
+Ideal functionality for AUTHDECODE:
+    Requester → ℱ: w
+    Notary → ℱ: ρ, sk
+    ℱ → Requester:
+        Let W = PRG(ρ)
+        Let p be the decoding of w wrt W
+        If decoding fails: abort
+        Else: return P, r, σ
+          where,
+            r is randomness
+            P = pack(p)
+            N = n/8
+            com_P = Com(P; r)
+            σ = Sign_sk(N || com_P),
+```
+
+## Protocol
+
+To start, we write a full real world protocol that instantiates the AUTHDECODE functionality. This makes use of various subprotocols, which we define below.
+```
+Impl for AUTHDECODE
+    Requester → Notary:
+        // Requester must commit to w before it learns the decoding
+            Send com_w = PolyCom(ck, w)
+    Requester ← Notary:
+        // ρ is sufficient to derive the full label set W
+            Send ρ
+    Requester → Notary:
+        // Reconstruct W and decode w into plaintext bits p
+            Let W = PRG(ρ)
+            Let p = Decode(w, W) or abort
+            Let P = pack(p)
+        // Commit to everything
+            Let com_p = PolyCom(ck, p)
+            Let com_P = PolyCom(ck, P)
+            Send (com_p, com_P)
+    Requester ↔ Notary:
+        // p is the decoding of w wrt W
+            Prove R_decode over (W₀, Δ, com_w, com_p; p, w)
+        // P is the packing of p
+            Prove R_pack over (com_P, com_p, ck; P, p)
+    Requester ← Notary:
+        // Everything succeeded, sign the length and commitment
+            Let N = n/8
+            Send σ = Sign_sk(N || com_P)
+```
+
+## Subprotocols
+
+Below we define the relations `R_decode`, `R_shiftcom`, and `R_pack`, and write ZK protocols for them.
+
+`R_decode` states that `p` is the decoding of `w` with respect to `W = W₀ || W₀ + (Δ, ..., Δ)`.
+
+**NOTE:** The protocol here seems suboptimal. In reality, we only need to prove the polynomial equality `W₀ = w - Δp`, where we have commitments to `w, p` and `W₀, Δ` are public. This should be very simple.
+```
+R_decode = {
+    (W₀, Δ, com_w, com_p, n; p, w) :
+      w = W₀ + pΔ
+    ∧ PolyVfyOpen(ck, com_w, w)
+    ∧ PolyVfyOpen(ck, com_p, p)
+}
+Protocol
+    Prover ↔ Verifier:
+        // Prove that p is binary
+            Prove R_binary over (ck, com_p, n; p)
+    Prover ← Verifier:
+            Sample and send a challenge point c
+            Let com_d = com_w - Δ·com_p
+    Prover → Verifier:
+        // Evaluate the LHS of the equation W₀ = w - Δp at c
+            Let W₀(X) = interp(W₀)
+            Let W₀c = W₀(c)
+            Let π = PolyWit(ck, W₀, (c, W₀c))
+    Verifier:
+        // Check the eval on W₀
+            Let com_W₀ = PolyCom(ck, W₀; 0)
+            Check PolyWitEval(ck, π, com_W₀, (c, W₀c))
```

Comment:
This step seems unnecessary.

Reply:
Yeah, so I didn't say this but this step is an optimization. The verifier could technicall compute `W₀c` itself, but we're trying to make verification cost as cheap as possible. So we let the prover do the eval for the verifier. Still though, `R_decode` requires 2 evals and it seems like overkill somehow

Reply:
Noted in latest version